src/log_impl.rs

```diff
@@ -116,6 +120,40 @@ pub struct Panic;
 
 pub struct Null;
 
+//Custom file logger which we will be able to manipulate the name
+#[derive(Debug)]
+pub struct CustomFileLog{
+    pub line_sep:  Cow<'static, str>,
+    pub file_suffix: String,
+    pub file_name: String,
+    pub curr_file_suffix: Mutex<String>,
+    pub is_writable: Mutex<bool>,
+    pub file_stream: Mutex<BufWriter<fs::File>>,
+}
+
+impl CustomFileLog{
+
+    pub fn get_suffix(file_suffix_pattern: &str) -> String {
+        chrono::Local::now().format(file_suffix_pattern).to_string()
```

Comment:
Ideally I'd like to avoid allocating a new string with `to_string()` on every log, but that might be unavoidable. Right now the log path is, for the most part, allocation free, and that helps with speed.

This seems like a hard problem to solve, though, so I'd be fine with leaving this in in this PR and possibly changing it later if we can find a solution.